src/core/validations/client_parity_tests/generate-messages.ts

```diff
@@ -0,0 +1,219 @@
+    fid: 11111,
+    network: FarcasterNetwork.TESTNET,
+    timestamp: getFixedFarcasterTime()
+  };
+
+  const signer = new NobleEd25519Signer(privateKey);
+
+  const result = await makeReactionAdd(reactionBody, dataOptions, signer);
+  return result._unsafeUnwrap();
+}
+
+async function createValidLinkAddMessage(): Promise<Message> {
+  const privateKey = getPrivateKey();
+
+  const linkBody: LinkBody = {
+    type: 'follow',
+    targetFid: 456,
+    displayTimestamp: undefined
+  };
+
+  const dataOptions = {
+    fid: 22222,
+    network: FarcasterNetwork.TESTNET,
+    timestamp: getFixedFarcasterTime()
+  };
+
+  const signer = new NobleEd25519Signer(privateKey);
+
+  const result = await makeLinkAdd(linkBody, dataOptions, signer);
+  return result._unsafeUnwrap();
+}
+
+async function createInvalidSignatureMessage(): Promise<Message> {
+  const validMessage = await createValidCastAddMessage();
+
+  // Corrupt the signature
+  const corruptedSignature = new Uint8Array(64);
+  corruptedSignature.fill(0);
+
+  return {
+    ...validMessage,
+    signature: corruptedSignature
+  };
+}
+
+async function createInvalidHashMessage(): Promise<Message> {
+  const validMessage = await createValidCastAddMessage();
+
+  // Corrupt the hash
+  const corruptedHash = new Uint8Array(20);
+  corruptedHash.fill(0);
+
+  return {
+    ...validMessage,
+    hash: corruptedHash
+  };
+}
+
+async function generateTestMessages() {
+  const outputDir = 'input_messages';
+
+  if (!fs.existsSync(outputDir)) {
+    fs.mkdirSync(path.join(__dirname, outputDir), { recursive: true });
+  }
+
+  const testInputs =
+    [
+      {
+        filename: 'valid_cast_add.bin',
+        message: await createValidCastAddMessage(),
+        isValid: true
+      },
+      {
+        filename: 'valid_user_data_add.bin',
+        message: await createValidUserDataAddMessage(),
+        isValid: true
+      },
+      {
+        filename: 'valid_reaction_add.bin',
+        message: await createValidReactionAddMessage(),
+        isValid: true
+      },
+      {
+        filename: 'valid_link_add.bin',
+        message: await createValidLinkAddMessage(),
+        isValid: true
+      },
+      {
+        filename: 'invalid_signature.bin',
+        message: await createInvalidSignatureMessage(),
+        isValid: false
+      },
+      {
+        filename: 'invalid_hash.bin',
+        message: await createInvalidHashMessage(),
+        isValid: false
+      },
+    ]
+
+  for (const testInput of testInputs) {
+    fs.writeFileSync(
+      path.join(__dirname, outputDir, testInput.filename),
+      Message.encode(testInput.message).finish()
+    );
+  }
+
+  console.log('All test message files generated successfully!');
+
+  const testInputsJson = testInputs.map((testInput) => {
+    return {
+      message_file: `${outputDir}/${testInput.filename}`,
+      is_valid: testInput.isValid
+    }
+  });
+
+  fs.writeFileSync(
+    path.join(__dirname, 'test_inputs.json'),
```

Comment:
Instead of separate files, can we encode to hex and keep it in the same file as the json?  Might be easier to manage. Don't feel too strongly though